rust/flexbuffers/src/reader/mod.rs

```diff
@@ -282,49 +292,61 @@ impl<'de> Reader<'de> {
             .buffer
             .get(self.address..end)
             .ok_or(Error::FlexbufferOutOfBounds)?;
+
         // `align_to` is required because the point of this function is to directly hand back a
         // slice of scalars. This can fail because Rust's default allocator is not 16byte aligned
         // (though in practice this only happens for small buffers).
         let (pre, mid, suf) = unsafe { slice.align_to::<T>() };
         if pre.is_empty() && suf.is_empty() {
-            Ok(mid)
+            // TODO(colindjk) is unsafe avoidable here?
```

Comment:
why does this seem unavoidable? it looks to me like you're casting `&[T]` to `*const [T]` and back... I'm guessing its a lifetime thing?

Note that this is a breaking change since previously you could hold onto the slice for as long as the backing buffer `'de` but now it lasts as long as `self` which may not live as long as `'de`, ie (`'self: 'de`). It may be unavoidable but at least we haven't promised stability yet.

Reply:
So I reviewed this and was able to avoid the extra `unsafe` code that I added. I think I was just mistaken earlier.